modal/functions.py

```diff
@@ -1155,10 +1154,37 @@ def _get_info(self):
     def _get_self_obj(self):
         return self._self_obj
 
+    @synchronizer.nowrap
+    def local(self, *args, **kwargs) -> Any:
+        # TODO(erikbern): it would be nice to remove the nowrap thing, but right now that would cause
+        # "user code" to run on the synchronicity thread, which seems bad
+        info = self._get_info()
+        if not info:
+            msg = (
+                "The definition for this function is missing so it is not possible to invoke it locally. "
+                "If this function was retrieved via `Function.lookup` you need to use `.call()`."
+            )
+            raise AttributeError(msg)
+
+        self_obj = self._get_self_obj()
+        if self_obj:
+            # This is a method on a class, so bind the self to the function
+            fun = info.raw_f.__get__(self_obj)
+        else:
+            fun = info.raw_f
+        return fun(*args, **kwargs)
+
     @synchronizer.nowrap
     def __call__(self, *args, **kwargs) -> Any:  # TODO: Generics/TypeVars
         if self._get_is_remote_cls_method():  # TODO(elias): change parametrization so this is isn't needed
-            return self.call(*args, **kwargs)
+            # TODO(erikbern): deprecate this soon too
+            return self.remote(*args, **kwargs)
+
+        deprecation_warning(
+            date(2018, 8, 16),
+            "Direct calling `f(...)` of local functions is deprecated. Use `f.local(...)` instead.",
```

Comment:
Since new users will run into this quite a bit, and we removed the "no remote calls" warning, maybe we should add some more text here to explain that function with name `x` is being called locally, and not remotely?

Reply:
Updated the warning message. Here's an example output:

```
erikbern@Eriks-MacBook-Air modal-examples % modal run 01_getting_started/hello_world.py
✓ Initialized. View app at https://modal.com/apps/ap-gCRHpKu0LRI1fTyteqLpGi
✓ Created objects.
├── 🔨 Created f.
└── 🔨 Created mount /Users/erikbern/modal-examples/01_getting_started/hello_world.py
/Users/erikbern/modal-examples/01_getting_started/hello_world.py:59: PendingDeprecationError: 2018-08-16: Calling Modal functions like `f(...)` is deprecated. Use `f.local(...)` if you want 
to call the function in the same Python process. Use `f.remote(...)` if you want to call the function in a Modal container in the cloud
  print(f(1000))
hello 1000
1000000
/Users/erikbern/modal-examples/01_getting_started/hello_world.py:62: PendingDeprecationError: 2018-08-16: `f.call(...)` is deprecated. It has been renamed to `f.remote(...)`
  print(f.call(1000))
hello 1000
1000000
hello 0
world 1
hello 2
world 3
hello 4
world 5
hello 6
world 7
hello 8
world 9
hello 10
world 11
hello 12
world 13
hello 14
world 15
hello 16
world 17
hello 18
world 19
2470
✓ App completed.
```